app/assets/javascripts/discourse/app/components/topic-footer-buttons.js

```diff
@@ -43,7 +43,12 @@ export default Component.extend({
 
   @discourseComputed("topic.isPrivateMessage")
   showNotificationsButton(isPM) {
-    return !isPM || this.siteSettings.enable_personal_messages;
+    return !isPM || this.canSendPms;
+  },
+
+  @discourseComputed("currentUser.groups.[]")
```

Comment:
It's not obvious at all that we need to listen to the current user's groups to have an accurate result for `allowPersonalMessages` 🤔 

How about we make it more explicit and change the signature of `allowPersonalMessages` to take these parameters `(isStaff, groups)`?

Reply:
Sometimes you're listening for both `currentUser.staff` and `currentUser.groups.[]` and other times, only for `currentUser.groups.[]`.

Reply:
I will listen to both instead :+1: And yeah good idea to add the additional parameter to make it more explicit.

Reply:
Oh I just reread:

> How about we make it more explicit and change the signature of allowPersonalMessages to take these parameters (isStaff, groups)?

This is more confusing to me...the function is checking whether the current user (if there is one) can send PMs based on the groups they are a member of, using `personal_message_enabled_groups` as a reference. I think the params as is are fine, but I will fix the computeds to check for `currentUser.staff` too

Reply:
I guess my point was that I find it hard to know that whenever you call `allowPersonalMessages` you have to listen to both `currentUser.staff` and `currentUser.groups.[]`.

Reply:
Yes I see now @ZogStriP . I am going to move this into current user service anyway at Joffrey's suggestion :+1:

Reply:
My bad, I wasn't clear enough. Thankfully @jjaffeux explained my idea better 👍